### PR TITLE
Fix Tifffile

### LINKED ISSUE #76147: pythonPackages.Tifffile is broken
**Describe the bug**
https://github.com/NixOS/nixpkgs/commit/845cecee49ebb5f923192956f05a5fe2d55c5ad6#diff-7011141acf30a6836812a1d147dbf58f added dependency on `imagecodecs-lite` and broke this package.

Maintainer information:
@FRidh @lebastr

### PR DESCRIPTION
**Pull request by tbenst**
#### Motivation for this change
closes #76147. Continuation of  #62755

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @FRidh @lebastr

### CODE REVIEW

File: pkgs/development/python-modules/tifffile/default.nix [hunk 1]
```diff
@@ -23,11 +24,12 @@ buildPythonPackage rec {
   '';
 
   propagatedBuildInputs = [ numpy ]
-    ++ lib.optional isPy27 [ futures enum34 pathlib ];
+    ++ lib.optional isPy27 [ futures enum34 pathlib ]
+    ++ lib.optional isPy3k [ imagecodecs-lite ];
```
Comment (JohnAZoidberg) [inline pkgs/development/python-modules/tifffile/default.nix]: Why is `imagecodecs-lite` only needed for py3k?

**Comment by JohnAZoidberg** [comment]
> broke this package

How was it broken and how can I verify that it is not broken anymore?

**Comment by jonringer** [comment]
cherry-picked to python-unstable [here](https://github.com/NixOS/nixpkgs/pull/76059/commits/7d279d7bf95598c65ff56d9c27bc0222750f7448). other commit was already applied

File: pkgs/development/python-modules/tifffile/default.nix (hunk 1 above)
Comment (tbenst) [reply pkgs/development/python-modules/tifffile/default.nix]: @JohnAZoidberg due to imagecodecs-lite being broken on python2: https://github.com/NixOS/nixpkgs/blob/6306baa8a2ad1f0db0994bc2a05278d0d3c8ffca/pkgs/top-level/python-packages.nix#L2627 and https://github.com/NixOS/nixpkgs/pull/62754#issuecomment-528038116.

**Pull request closed without merging**